src/integrationTest/kotlin/com/sourcegraph/cody/util/CodyIntegrationTextFixture.kt

```diff
@@ -49,6 +49,13 @@ open class CodyIntegrationTextFixture : BasePlatformTestCase() {
           logger.warn("Error shutting down session", x)
         }
       }
+      CodyAgentService.getInstance(myFixture.project).apply {
+        try {
+          dispose()
```

Comment:
This is not shutting down agent, but finishing current fixup session.
IMHO it makes sense to wait for such finish, but if it is not able to finish itself in a given time window it should be a test failure.
So instead of disposing it we should wait and check there.